pywps/app/Service.py

```diff
@@ -61,26 +75,37 @@ def __init__(self, processes: Sequence = [], cfgfiles=None, preprocessors: Optio
             if not LOGGER.handlers:
                 LOGGER.addHandler(logging.NullHandler())
 
-    def get_capabilities(self, wps_request, uuid):
-
-        response_cls = response.get_response("capabilities")
-        return response_cls(wps_request, uuid, version=wps_request.version, processes=self.processes)
-
-    def describe(self, wps_request, uuid, identifiers):
-
-        response_cls = response.get_response("describe")
-        return response_cls(wps_request, uuid, processes=self.processes,
-                            identifiers=identifiers)
+    @operations("status")
+    def get_status(self, wps_request, uuid):
```

Comment:
Shouldn't this be `GetStatus`? 
This is the name of the WPS operation: http://opengeospatial.github.io/e-learning/wps/text/operations.html

Reply:
Hello @fmigneault ,

Thanks for feedbacks.

I was not aware of GetStatus, I am focused on WPS 1.0.0 spec. I used request=status to avoid complex change in code. Initially I used a separate url than the `[server] url` but it require more setup to ensure wsgi to expose the new url path, and need some change on how we dispatch urls within the code, thus I dropped the idea and switched to the most convenient implementation to start the discussion. I also considered to use service=something. I will be happy to get some suggestion about this url, the only requirement is that the request need the status uuid to retrieve the status within the database.

As far as I understand GetStatus, it's seems it not the same as the WPS status file required in WPS 1.0.0 thus they cannot be merged together. Nevertheless, the GetStatus request can be implemented on top of this patch suite, all data required for generating the GetStatus outputs is stored in the database in this patch suite. We can also implement the GetResult.

Reply:
Sorry for the long delay. 

`service=...` should be avoided as it is (normally) intended to redirect to other services such as WMS, WCS, etc.
Currently, PyWPS does not have other services running under the same webapp, but a server using a PyWPS behind the scene could filter this parameter to perform some redirections to other endpoints. 

The idea of `GetStatus` is to retrieve the current status of the job, regardless of where the status is stored.
However, the full XML details of the status can always be retrieved from `statusLocation` provided in the execute response. This location does not necessarily have to be a file, but this is what PyWPS currently uses. That location does not have to be on the same server either. It could point at a remote instance or an S3 bucket for example.

If `GetStatus` is needed, I think the right approach would be to employ WPS 2.0. It would be preferable to support (relatively) newer revisions than patch older ones with unofficial requests.

Reply:
Ok, do you have some idea of which url I should use ?

I will remove the decorator @operations that can be confusion within the code.

Reply:
It should be a configurable setting to allow instances to adapt as needed.
Similar to how the `[server] outputurl` can be configured, it could be a `[server] statusurl` or similar.
This way, the server can decide where to host status files (or not if the setting is left empty by default).

Reply:
There is no need of file for dynamic status, thus the url does not need a mapping to a file on the file system. I will keep an option to keep the file on file system with same behaviour as before, but the url that I talk about will be provided by wsgi application directly, not through apache, in the same manner as DescribeProcess request for instance, or as we will have to implement for WPS 2.0.0, i.e. GetStatus.

Currently this url look like:
```
http://hostname/path/to/wps?service=WPS&request=status&uuid=SOMEUUID
```
where `http://hosname/pat/to/wps` is the setting provided by `[server] url`

If we want to use another unrelated path than the one above, it require that user configure a second wsgi application.

One option is to use a sub-path on top of the one above as instance it can be:
```
http://hostname/path/to/wps/status?uuid=SOMEUUID
```
or
```
http://hostname/path/to/wps/status/SOMEUUID
```
Another option is to not bind the WPS with the url option `[server] url` directly but having wps service binded to sub-path that way in the configuration above the wps may by bound to a path like:
```
http://hostname/path/to/wps/wps?service=WPS&request=GetCapabilities
```
having status urls and wps request url at the same level.
In this case we may have following configurations:
```
[server]
rooturl=http://hostname/path/to/wsgi
status_sub_path=custom_status_path
wps_sub_path=custom_wps_path
```
that will produce the following urls for WPS request:
```
http://hostname/path/to/wsgi/custom_wps_path?service=WPS&request=GetCapabilities
```
and
```
http://hostname/path/to/wsgi/custom_status_path?uuid=someuuid
```
The last options is my favorite compromise. default value for wps_sub_path can be 'wps' and the default status_sub_path can be 'status'